example/parsec/src/parsec.c

```diff
@@ -87,40 +132,27 @@
                                                                          \
   END_OF_STATEMENTS
 
+// -----------------------------------------------------------------------
 trait_ParsecLibrary(String);
 impl_ParsecLibrary(String);
 
 // -----------------------------------------------------------------------
 #define S String
 
-#define parseTest(p, input)                                              \
-  GENERIC_PARSECRUNNER(String, p).parseTest(p, input)
-
-#define parse(p, state)                                                  \
-  GENERIC_PARSECRUNNER(String, p).runParsec(p, state)
-
 // -----------------------------------------------------------------------
-#define DO()                                                             \
-  g_bind((_s0_, _cok_, _cerr_, _eok_, _eerr_), *args);                   \
-  __auto_type _s_ = _s0_;
-
-#define SCAN(...) CAT(SCAN, VARIADIC_SIZE(__VA_ARGS__))(__VA_ARGS__)
-#define SCAN1(_p_)                                                       \
-  __auto_type TMPID = parse(_p_, _s_);                                   \
-  if (!TMPID.result.success) {                                           \
-    __auto_type _err_ = (TMPID.consumed ? _cerr_ : _eerr_);              \
-    return fn_apply(_err_, TMPID.result.err, TMPID.state);               \
-  }                                                                      \
-  _s_ = TMPID.state;
-#define SCAN2(_p_, _x_)                                                  \
-  SCAN1(_p_);                                                            \
-  __auto_type _x_ = TMPID.result.ok;
+fn(abcFn, UnParserArgs(S, Array(char))) {
+  __auto_type char1 = trait(ParsecChar(S)).char1;
+  DO() {
+    SCAN(char1('a'), a);
+    SCAN(char1('b'), b);
+    SCAN(char1('c'), c);
+    RETURN(g_array(char, a, b, c));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Skipping configuration 'CPARSEC_STREAM_TYPE' since the value of 'CPARSEC_STREAM_TYPE' is unknown. Use -D if you want to check it. You can use -U to skip it explicitly.](https://app.codacy.com/manual/mori0091/cparsec3/pullRequest?prid=5138570)